src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -189,6 +232,7 @@ void DefaultOTARequestorDriver::UpdateNotFound(UpdateNotFoundReason reason, Syst
         if ((GetNextProviderLocation(providerLocation, listExhausted) != true) || (listExhausted == true))
         {
             willTryAnotherQuery = false;
+            status              = CHIP_ERROR_MAX_RETRY_EXCEEDED;
```

Comment:
Same comment here, this status is not indicative of what actually is happening. Perhaps you are treating either case the same way from the calling side?

Reply:
It indicates that we cannot schedule another retry, either due to current provider max retry account exceeded or there is no additional providers in the provider list to retry with.

Reply:
So perhaps for the `Busy` case, you want to know list has been exhausted and therefore, do not query further, but there are other places where you now call `ScheduleQueryRetry(true)` but reaching the end of the list results in nothing happening when perhaps, do we want to try?

The point is, we probably want to differentiate the true case of we've reached max retries vs we've reached the end of the list (by returning different CHIP_ERROR in `ScheduleQueryRetry`) such that the caller can then determine what it wants to do next.

Reply:
Created CHIP_ERROR_PROVIDER_LIST_EXHAUSTED  to be used when provider list is exhausted.